api/hasura/cron/dailyUpdate.ts

```diff
@@ -0,0 +1,203 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import dedent from 'dedent';
+import { DateTime, DurationObjectUnits } from 'luxon';
+
+import { gql } from '../../../api-lib/Gql';
+import { sendSocialMessage } from '../../../api-lib/sendSocialMessage';
+import { verifyHasuraRequestMiddleware } from '../../../api-lib/validate';
+import { pending_token_gifts_select_column } from '../../../src/lib/gql/__generated__/zeusAdmin';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  const yesterday = DateTime.now().minus({ days: 1 }).toISO();
+  try {
+    const updateResult = await gql.q('query')({
+      epochs: [
+        {
+          where: {
+            end_date: { _gt: 'now()' },
+            start_date: { _lt: 'now()' },
+            ended: { _eq: false },
+          },
+        },
+        {
+          number: true,
+          start_date: true,
+          end_date: true,
+
+          circle: {
+            organization: {
+              name: true,
+            },
+            id: true,
+            name: true,
+            token_name: true,
+            discord_webhook: true,
+            telegram_id: true,
+
+            __alias: {
+              optOuts: {
+                users_aggregate: [
+                  { where: { non_receiver: { _eq: true } } },
+                  {
+                    aggregate: { count: [{}, true] },
+                  },
+                ],
+              },
+              receiversTotal: {
+                users_aggregate: [
+                  { where: { non_receiver: { _eq: false }, role: { _lt: 2 } } },
```

Comment:
same here with total receiver. I added the role filter to exclude the coordinape user.

Reply:
very observant to exclude coordinape user